### PR TITLE
support audit_metadata in widget

### PR DESCRIPTION
**Pull request by ebrehault**

### CODE REVIEW

File: libs/search-widget/src/widgets/search-widget/SearchBar.svelte [hunk 1]
```diff
@@ -85,6 +85,7 @@
   export let json_schema = '';
   export let vectorset = '';
   export let chat_placeholder = '';
+  export let audit_metadata = '';
```
Comment (operramon) [inline libs/search-widget/src/widgets/search-widget/SearchBar.svelte:88]: should also be added to popup and chat widgets ?

File: libs/common/src/lib/search-widget/search-widget.service.ts [hunk 2]
```diff
@@ -194,6 +194,7 @@ export class SearchWidgetService {
 
         let baseSnippet = `<${tagName}${theme}\n  knowledgebox="${kb.id}"`;
         baseSnippet += `\n  ${zone}${features}${prompt}${systemPrompt}${rephrasePrompt}${ragProperties}${ragImagesProperties}${placeholder}${chatPlaceholder}${notEnoughDataMessage}${askToResource}${maxTokens}${maxParagraphs}${queryPrepend}${generativeModel}${vectorset}${filters}${preselectedFilters}${privateDetails}${backend}${jsonSchema}`;
+        baseSnippet += `\n  audit_metadata="{\\"config\\":\\"${currentConfig.id}\\"}"`;
```
Comment (operramon) [inline libs/common/src/lib/search-widget/search-widget.service.ts:197]: is this line added on purpose ?

**Review by operramon — APPROVED** [review]

File: libs/search-widget/src/widgets/search-widget/SearchBar.svelte (hunk 1 above)
Comment (ebrehault) [reply libs/search-widget/src/widgets/search-widget/SearchBar.svelte:88]: right, fixed!

File: libs/common/src/lib/search-widget/search-widget.service.ts (hunk 2 above)
Comment (ebrehault) [reply libs/common/src/lib/search-widget/search-widget.service.ts:197]: Yes, by default we put the search config id in the audit metadata, and the users are free to add more manually after pasting the snippet if they want to.

**Pull request merged**